src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -647,15 +647,41 @@ protected void computeChildren(final ChildObserver<P> observer, final TaskListen
     }
 
     private void scheduleBuild(BranchProjectFactory<P, R> factory, final P item, SCMRevision revision,
-                               TaskListener listener, String name, Cause[] cause, Action... actions) {
+                               TaskListener listener, String name, Cause[] causes, Action... actions) {
         if (!isBuildable()) {
             listener.getLogger().printf("Did not schedule build for branch: %s (%s is disabled)%n",
                     name, getDisplayName());
             return;
         }
-        Action[] _actions = new Action[actions.length + 1];
-        _actions[0] = new CauseAction(cause);
-        System.arraycopy(actions, 0, _actions, 1, actions.length);
+        // JENKINS-48090 see Queue.Item.getCauses() which only operates on the first CauseAction
```

Comment:
That sounds at all that it was bad idea to wrap causes into action in core...

Reply:
Well we could argue that this is actually a bug in core... namely that it only unwraps the first `CauseAction` but I do not want to get into a fight on that